From be34d026f44389745e29a7da977f84712066d3d0 Mon Sep 17 00:00:00 2001
From: Bernd Kuhls <bernd@kuhls.net>
Date: Sat, 15 Nov 2025 19:41:09 +0100
Subject: [PATCH] Fixed compilation problems with PATH_MAX
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

In file included from ../src/cpxm.c:38:
../src/libdvdcss.h:65:26: error: ‘PATH_MAX’ undeclared here (not in a function)
   65 |     char   psz_cachefile[PATH_MAX];

In file included from ../src/error.c:26:
../src/libdvdcss.h:65:26: error: ‘PATH_MAX’ undeclared here (not in a function)
   65 |     char   psz_cachefile[PATH_MAX];

In file included from ../src/libdvdcpxm.c:47:
../src/libdvdcss.h:65:26: error: ‘PATH_MAX’ undeclared here (not in a function)
   65 |     char   psz_cachefile[PATH_MAX];

Upstream: https://code.videolan.org/videolan/libdvdcss/-/merge_requests/24

Signed-off-by: Bernd Kuhls <bernd@kuhls.net>
---
 src/libdvdcss.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/libdvdcss.h b/src/libdvdcss.h
index f9d36ba..997af34 100644
--- a/src/libdvdcss.h
+++ b/src/libdvdcss.h
@@ -26,6 +26,11 @@
 
 #include <limits.h>
 
+#include <config.h>
+#ifdef HAVE_SYS_PARAM_H
+#   include <sys/param.h>
+#endif
+
 #include "dvdcss/dvdcss.h"
 #include "cpxm.h"
 #include "css.h"
-- 
2.47.3

